docs/how-to/debugging-guide.rst

```diff
@@ -0,0 +1,245 @@
+
+   .. code:: shell
+
+      cat some_name_here_*.log > some_name_here.log
+
+#. Run the RCCL Replayer using the following command. Replace ``<numProcesses>`` with the number of MPI processes to 
+   run, ``</path/to/logfile>`` with the path to the collective log file generated during 
+   the RCCL runs, and ``<numGpusPerMpiRank>`` with the number of GPUs per MPI rank used in the application.
+
+   .. code:: shell
+
+      mpirun -np <numProcesses> ./rcclReplayer </path/to/logfile> <numGpusPerMpiRank>
+
+   In a multi-node application environment, you can replay the collective logs on multiple nodes
+   using the following command:
+
+   .. code:: shell
+
+      mpirun --hostfile <path/to/hostfile.txt> -np <numProcesses> ./rcclReplayer </path/to/logfile> <numGpusPerMpiRank>
+
+   .. note::
+
+      Depending on the MPI library you are using, you might need to modify the ``mpirun`` command.
+
+Troubleshooting
+=============================
+
+Use the following troubleshooting techniques to attempt to isolate the issue.
+
+*  Build or run the develop branch version of RCCL and see if the problem persists.
+*  Try an earlier RCCL version (minor or major).
+*  If you recently changed the ROCm runtime configuration, KFD/driver, or compiler,
+   rerun the test with the previous configuration.
+
+.. _analyze-performance-info:
+
+Analyzing performance issues
+=============================
+
+If the issues involve performance issues in an e2e workload, try the following 
+microbenchmarks and collect the results. Follow the instructions in the subsequent sections
+to run these benchmarks and provide the results to the support team.
+
+*  TransferBench
+*  RCCL Unit Tests
+*  rccl-tests
+  
+Collect the TransferBench data
+---------------------------------
+
+TransferBench allows you to benchmark simultaneous copies between
+user-specified devices. For more information, 
+see :doc:`the TransferBench documentation <transferbench:index>`.
+
+To collect the TransferBench data, follow these steps:
+
+#. Clone the TransferBench Git repository.
+
+   .. code:: shell
+
+      git clone https://github.com/ROCm/TransferBench.git 
+
+#. Change to the new directory and build the component.
+
+   .. code:: shell
+
+      cd TransferBench
+      make
+
+#. Run the TransferBench utility with the following parameters and save the results.
+
+   .. code:: shell
+
+      USE_FINE_GRAIN=1 GFX_UNROLL=2 ./TransferBench a2a 64M 8
+
+Collect the RCCL microbenchmark data
+-------------------------------------
+
+To use the RCCL tests to collect the RCCL benchmark data, follow these steps:
+
+#. Disable NUMA auto-balancing using the following command:
+
+   .. code:: shell
+
+      sudo sysctl kernel.numa_balancing=0
+
+   Run the following command to verify the setting. The expected output is ``0``.
+
+   .. code:: shell
+
+      cat /proc/sys/kernel/numa_balancing
+
+#. Build MPI, RCCL, and rccl-tests. To download and install MPI, see the `MPI site <https://www.mpich.org/>`_.
```

Comment:
instead of `MPI site`, can we suggest both OpenMPI (https://www.open-mpi.org/software/ompi/v5.0/) and MPICH (https://www.mpich.org/)